homeassistant/components/devolo_home_network/update.py

```diff
@@ -0,0 +1,134 @@
+"""Platform for update integration."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from devolo_plc_api.device import Device
+from devolo_plc_api.device_api import UpdateFirmwareCheck
+from devolo_plc_api.exceptions.device import DevicePasswordProtected, DeviceUnavailable
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityDescription,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, REGULAR_FIRMWARE
+from .entity import DevoloCoordinatorEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class DevoloUpdateRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    latest_version: Callable[[UpdateFirmwareCheck], str]
+    update_func: Callable[[Device], Awaitable[bool]]
+
+
+@dataclass
+class DevoloUpdateEntityDescription(
+    UpdateEntityDescription, DevoloUpdateRequiredKeysMixin
+):
+    """Describes devolo switch entity."""
+
+
+UPDATE_TYPES: dict[str, DevoloUpdateEntityDescription] = {
+    REGULAR_FIRMWARE: DevoloUpdateEntityDescription(
+        key=REGULAR_FIRMWARE,
+        device_class=UpdateDeviceClass.FIRMWARE,
+        entity_category=EntityCategory.CONFIG,
+        latest_version=lambda data: data.new_firmware_version.split("_")[0],
+        update_func=lambda device: device.device.async_start_firmware_update(),  # type: ignore[union-attr]
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Get all devices and sensors and setup them via config entry."""
+    device: Device = hass.data[DOMAIN][entry.entry_id]["device"]
+    coordinators: dict[str, DataUpdateCoordinator[Any]] = hass.data[DOMAIN][
+        entry.entry_id
+    ]["coordinators"]
+
+    entities: list[DevoloUpdateEntity] = []
+    entities.append(
+        DevoloUpdateEntity(
+            entry,
+            coordinators[REGULAR_FIRMWARE],
+            UPDATE_TYPES[REGULAR_FIRMWARE],
+            device,
+        )
+    )
+    async_add_entities(entities)
```

Comment:
```suggestion
    async_add_entities(
        DevoloUpdateEntity(
            entry,
            coordinators[REGULAR_FIRMWARE],
            UPDATE_TYPES[REGULAR_FIRMWARE],
            device,
        )
    )
```

Reply:
It seems, that this doesn't work that way as async_add_entities expects an iterable.

Reply:
Updated the suggestion, wrap it in []